Sources/DDGSync/DDGSync.swift

```diff
@@ -161,31 +162,83 @@ public class DDGSync: DDGSyncing {
         do {
             try await dependencies.account.deleteAccount(account)
             try dependencies.secureStore.removeAccount()
-            updateIsAuthenticated()
+            updateAuthState()
         } catch {
             try handleUnauthenticated(error)
         }
     }
 
     // MARK: -
 
-    let persistence: LocalDataPersisting
     let dependencies: SyncDependencies
 
-    init(persistence: LocalDataPersisting, dependencies: SyncDependencies) {
-        self.persistence = persistence
+    init(dataProvidersSource: DataProvidersSource, dependencies: SyncDependencies) {
+        self.dataProvidersSource = dataProvidersSource
         self.dependencies = dependencies
-        self.isAuthenticated = (try? dependencies.secureStore.account()?.token) != nil
+        self.authState = .inactive
+
+        // do not start sync upon initialization
+        updateAuthState(startSyncIfNeeded: false)
     }
 
-    private func updateIsAuthenticated() {
-        isAuthenticated = (try? dependencies.secureStore.account()?.token) != nil
+    private func updateAuthState(startSyncIfNeeded: Bool = true) {
+        let previousState = authState
+        authState = (try? dependencies.secureStore.account()?.state) ?? .inactive
+
+        if previousState == .inactive && authState != .inactive {
+            let providers = dataProvidersSource?.makeDataProviders() ?? []
+            let syncQueue = SyncQueue(dataProviders: providers, dependencies: dependencies)
+
+            syncQueueCancellable = syncQueue.isSyncInProgressPublisher
+                .sink(receiveCompletion: { [weak self] _ in
+                    self?.isSyncInProgressSubject.send(false)
+                }, receiveValue: { [weak self] isInProgress in
+                    self?.isSyncInProgressSubject.send(isInProgress)
+                })
+
+            startSyncCancellable = dependencies.scheduler.startSyncPublisher
+                .sink { [weak self] in
+                    guard let self else {
+                        return
+                    }
+                    Task {
+                        if self.authState == .active {
+                            await syncQueue.startSync()
+                        } else {
+                            await syncQueue.setUpAndStartFirstSync()
+                        }
+                    }
+                }
+
+            syncDidFinishCancellable = syncQueue.syncDidFinishPublisher
+                .sink { [weak self] result in
+                    if case .success = result {
+                        self?.updateAuthState()
```

Comment:
Is this needed?

Reply:
I figured out why it was there. This is a leftover from when I tried to have a single `startSync` method to handle both initial and regular sync. It was a bad decision and I’ve since switched to 2 methods (`startSync` and `setUpAndStartFirstSync`) but have never deleted this subscription. It’s now gone though :)